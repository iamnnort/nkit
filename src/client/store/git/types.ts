import { CommonStore } from '../../lib/store/types';

export enum ActionTypes {
  LOAD_RELEASES_REQUEST = '@@git/LOAD_RELEASES_REQUEST',
  LOAD_RELEASES_SUCCESS = '@@git/LOAD_RELEASES_SUCCESS',
  LOAD_RELEASES_FAILURE = '@@git/LOAD_RELEASES_FAILURE',
}

export interface Git extends CommonStore {
  releases: GitRelease[];
}

export interface GitRelease {
  id: number;
  name: string;
  tag_name: string;
  html_url: string;
}
