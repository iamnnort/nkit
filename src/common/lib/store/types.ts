export interface CommonStore {
  isInitialLoaded: boolean;
  isLoading: boolean;
  error: string;
}

export enum CommonError {
  FAILED_REQUEST = 'Loading was failed',
}
