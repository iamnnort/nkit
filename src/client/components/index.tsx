import Icon from './Icon/Icon';
import Loader from './Loader/Loader';

export { Icon, Loader };
