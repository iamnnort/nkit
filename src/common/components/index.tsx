import Icon from '@common/components/Icon/Icon';
import Loader from '@common/components/Loader/Loader';

export { Icon, Loader };
