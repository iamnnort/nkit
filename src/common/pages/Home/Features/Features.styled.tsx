import styled, { icon } from '../../../theme/styled-components';

export const Features = styled.section``;

export const FeaturesTitle = styled.div`
  font-size: ${(props) => props.theme.mediumFontSize};
  font-weight: bold;
`;

export const FeaturesSearch = styled.input`
  width: 100%;

  margin-top: ${(props) => props.theme.smallPadding};
  padding: 5px;
  border: 1px solid ${(props) => props.theme.textColor};
`;

export const FeaturesList = styled.div`
  margin-top: ${(props) => props.theme.smallPadding};
  padding-left: 5px;
`;

export const FeaturesListItem = styled('div')<{
  isSelected: boolean;
}>`
  display: flex;
  align-items: center;

  svg {
    margin-right: 10px;
  }

  ${(props) => props.isSelected && icon({ color: props.theme.brandColor })}
`;
