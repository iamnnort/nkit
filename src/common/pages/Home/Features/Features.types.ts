export interface Feature {
  value: string;
}

export interface ComponentProps {
  features: Feature[];
}
