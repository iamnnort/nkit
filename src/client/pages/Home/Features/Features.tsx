import * as React from 'react';

import { Features, FeaturesTitle, FeaturesList, FeaturesListItem } from './Features.styled';
import { ComponentProps } from './Features.types';
import { Icon } from '../../../components';

const ReleasesComponent: React.FC<ComponentProps> = ({ features }) => {
  return (
    <Features>
      <FeaturesTitle>{'Features'}</FeaturesTitle>
      <FeaturesList>
        {features.map((feature) => (
          <FeaturesListItem key={feature.value}>
            <Icon type={'checkMark'} />
            {feature.value}
          </FeaturesListItem>
        ))}
      </FeaturesList>
    </Features>
  );
};

export default ReleasesComponent;
