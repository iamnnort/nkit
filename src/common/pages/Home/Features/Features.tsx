import * as React from 'react';
import { useTranslation } from 'react-i18next';

import { Features, FeaturesSearch, FeaturesTitle, FeaturesList, FeaturesListItem } from './Features.styled';
import { ComponentProps } from './Features.types';
import { Icon } from '../../../components';

const FeaturesComponent: React.FC<ComponentProps> = ({ features }) => {
  const { t } = useTranslation();

  const [searchValue, setSearchValue] = React.useState('re');

  const filteredFeatures = React.useMemo(() => {
    const regex = new RegExp(searchValue, 'i');

    return features.map(feature => {
      return {
        ...feature,
        isSelected: regex.test(feature.value),
      }
    })
  }, [features, searchValue]);

  const handleChangeSearch = React.useCallback((event) => {
    setSearchValue(event.target.value);
  }, [setSearchValue]);

  return (
    <Features>
      <FeaturesTitle>{t('general:features')}</FeaturesTitle>
      <FeaturesSearch value={searchValue} onChange={handleChangeSearch} />
      <FeaturesList>
        {filteredFeatures.map((feature) => (
          <FeaturesListItem key={feature.value} isSelected={feature.isSelected}>
            <Icon type={'checkMark'} />
            {feature.value}
          </FeaturesListItem>
        ))}
      </FeaturesList>
    </Features>
  );
};

export default FeaturesComponent;
