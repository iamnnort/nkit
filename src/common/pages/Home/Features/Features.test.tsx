import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import FeaturesComponent from './Features';
import { FeaturesSearch, FeaturesListItem, FeaturesTitle } from './Features.styled';
import { ComponentProps } from './Features.types';
import { Icon } from '../../../components';

describe('FeaturesComponent', () => {
  let wrapper: ReactWrapper;

  const props: ComponentProps = {
    features: [
      {
        value: 'React - Main library',
      },
    ],
  };

  const helpers = {
    search(value: string) {
      const input = wrapper.find(FeaturesSearch);
      input.simulate('change', { target: { value } });
    }
  }

  beforeEach(async () => {
    wrapper = mount(<FeaturesComponent {...props} />);
  });

  it(`should not regret`, () => {
    expect(wrapper).toMatchSnapshot();
  });

  it(`should render FeaturesListItem ${props.features.length} time(s)`, () => {
    expect(wrapper.find(FeaturesListItem).length).toEqual(props.features.length);
  });

  it(`should render FeaturesListItem depends on a search string`, () => {
    wrapper.setProps({
      features: [
        {
          value: 'React - Main library',
        },
        {
          value: 'Redux - Global store',
        },
      ],
    });

    helpers.search('redux');

    expect(
      wrapper.find(FeaturesListItem).filterWhere((featuresListItem) => featuresListItem.prop('isSelected') === true)
        .length
    ).toEqual(1);
  });

  describe('should have FeaturesSearch which', () => {
    it(`should have a default value`, () => {
      expect(wrapper.find(FeaturesSearch).prop('value')).toEqual('re');
    });

    it(`should have a new value after firing onChange event`, () => {
      helpers.search('new value');

      expect(wrapper.find(FeaturesSearch).prop('value')).toEqual('new value');
    });
  });

  describe('should have FeaturesTitle which', () => {
    it(`should set a title`, () => {
      expect(wrapper.find(FeaturesTitle).text()).toEqual('general:features');
    });
  });

  describe('should have FeaturesListItem which', () => {
    it(`should have an icon`, () => {
      expect(wrapper.exists(Icon)).toEqual(true);
    });
  });
});
