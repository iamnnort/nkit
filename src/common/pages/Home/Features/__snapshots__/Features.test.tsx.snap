// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturesComponent should not regret 1`] = `
.c0 {
  font-weight: bold;
}

.c1 {
  width: 100%;
  padding: 5px;
  border: 1px solid;
}

.c2 {
  padding-left: 5px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 svg {
  margin-right: 10px;
}

.c3 svg {
  display: inherit;
}

<FeaturesComponent
  features={
    Array [
      Object {
        "value": "React - Main library",
      },
    ]
  }
>
  <styled.section>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "bDWFJH",
            "rules": Array [],
          },
          "displayName": "styled.section",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <section
        className=""
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c0",
                  "rules": Array [
                    "
  font-size: ",
                    [Function],
                    ";
  font-weight: bold;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c0"
            >
              general:features
            </div>
          </StyledComponent>
        </styled.div>
        <styled.input
          onChange={[Function]}
          value="re"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  width: 100%;

  margin-top: ",
                    [Function],
                    ";
  padding: 5px;
  border: 1px solid ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "styled.input",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onChange={[Function]}
            value="re"
          >
            <input
              className="c1"
              onChange={[Function]}
              value="re"
            />
          </StyledComponent>
        </styled.input>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": false,
                  "lastClassName": "c2",
                  "rules": Array [
                    "
  margin-top: ",
                    [Function],
                    ";
  padding-left: 5px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c2"
            >
              <styled.div
                isSelected={true}
                key="React - Main library"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-ifAKCX",
                        "isStatic": false,
                        "lastClassName": "c3",
                        "rules": Array [
                          "
  display: flex;
  align-items: center;

  svg {
    margin-right: 10px;
  }

  ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-ifAKCX",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  isSelected={true}
                >
                  <div
                    className="c3"
                  >
                    <default
                      type="checkMark"
                    >
                      <MockComponent
                        originalComponent="Icon"
                        type="checkMark"
                      />
                    </default>
                    React - Main library
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
      </section>
    </StyledComponent>
  </styled.section>
</FeaturesComponent>
`;
